File: backend/tiggle/src/main/java/com/side/tiggle/domain/reaction/repository/ReactionRepository.java
package com.side.tiggle.domain.reaction.repository;

import com.side.tiggle.domain.reaction.model.Reaction;
import com.side.tiggle.domain.reaction.model.ReactionType;
import org.springframework.data.jpa.repository.JpaRepository;

public interface ReactionRepository extends JpaRepository<Reaction, Long> {

    Reaction findByTxIdAndSenderId(long txId, long senderId);
    int countAllByTxIdAndType(long txId, ReactionType type);
    int deleteByTxIdAndSenderId(long txId, long senderId);

}

File: backend/tiggle-root/tiggle/src/main/kotlin/com/side/tiggle/domain/reaction/repository/ReactionRepository.kt
package com.side.tiggle.domain.reaction.repository

import com.side.tiggle.domain.reaction.model.Reaction
import com.side.tiggle.domain.reaction.model.ReactionType
import org.springframework.data.jpa.repository.JpaRepository

interface ReactionRepository: JpaRepository<Reaction, Long> {
    fun findByTxIdAndSenderId(txId: Long, senderId: Long): Reaction?
    fun countAllByTxIdAndType(txId: Long, type: ReactionType): Int
    fun deleteByTxIdAndSenderId(txId: Long, senderId: Long): Int
}
